### PR TITLE
Remove unused code for various endpoints (two of which don't exist).

### PR DESCRIPTION
**Pull request by chrisbobbe**
Fixes: #3540 

I found a few more endpoints that @timabbott mentioned in https://github.com/zulip/zulip/issues/10044 as candidates for cleaning up, not mentioned in #3540. Also, while all of these endpoints are currently unused in the mobile app, it's likely that that will change for some of them, and I've marked them in the commit messages.

Two endpoints (`GET users/me/user_groups` and `GET realm/user_groups/${id}`) just don't exist and never have.

### CODE REVIEW

**Comment by agrawal-d** [comment]
> Two endpoints (`GET users/me/user_groups` and `GET realm/user_groups/${id}`) just don't exist and never have.

I wonder how those got added in the first place 😂.

**Comment by gnprice** [comment]
Thanks @chrisbobbe!

All of these removals LGTM; feel free to merge after editing the "don't merge" from the commit messages.

Specific thoughts:

    "I believe the app doesn't support editing these organization
    settings. Further, it should be just getting these data from
    /register with the right event_types value, so it's probably a
    mobile bug if we're accessing these:"
    
    This endpoint is included, but I don't think it's for fetching data,
    so I wanted to confirm that we should remove this code.

It's not for fetching data, but it is for editing some organization settings, and we don't have any UI for editing org settings. Eventually perhaps we'll want to build one, or at least some fragments of one, but in general I think it's a low priority. (This applies to several commits.)

Deleting a stream is a more likely nearer-term thing to add, and it's also a natural pair with `createStream` which we do use. So maybe that would be the one of these to keep around. It's also cheap to add later when we are going to use it, though, so perhaps best to stick to routes we're using or have concrete plans to use.

    toggleMuteStream exists, and is already being used. Favor that one,
    since other code that we do use follows that pattern; see
    togglePinStream, toggleStreamNotifications, etc.
    
    muteTopic and unmuteTopic is an exception; perhaps we would like to
    refactor that into a toggleMuteTopic.

I agree on both points. One thing to add: "toggle" is a bad name for what all those functions do. Typically in software use it means specifically to set a boolean flag to the opposite of whatever value it currently has; here, we've been using it inappropriately to just mean setting the flag to a value.

One good name might be `setTopicMute`. For `toggleMuteStream` and friends, a second independent improvement would be to take the opportunity to have fewer idiosyncratic names of our own, and use more names verbatim from the API -- so, together, perhaps a `setSubscriptionProperty` that takes a `property: 'is_muted' | 'pin_to_top' | ...` and a `value: boolean`.

**Comment by chrisbobbe** [comment]
> I agree on both points. One thing to add: "toggle" is a bad name for what all those functions do. Typically in software use it means specifically to set a boolean flag to the opposite of whatever value it currently has; here, we've been using it inappropriately to just mean setting the flag to a value.

Yep, I agree. It seems odd that we've used "toggle" in any of these, given that the server isn't handling any of them as a toggle. On looking a bit closer than I did before, I think one good theory for the origin of these names is that the names were influenced by the UI; the functions named with 'toggle' are called when a RN toggle switch is pressed; `muteTopic` and `unmuteTopic` aren't named with 'toggle', and it turns out that they aren't associated with toggle switches. 

> One good name might be `setTopicMute`.

Makes sense.

> For `toggleMuteStream` and friends, a second independent improvement would be to take the opportunity to have fewer idiosyncratic names of our own, and use more names verbatim from the API -- so, together, perhaps a `setSubscriptionProperty` that takes a `property: 'is_muted' | 'pin_to_top' | ...` and a `value: boolean`.

This sounds good too.

**Comment by chrisbobbe** [comment]
OK, I've pushed two commits that get us those `setTopicMute` and `setSubscriptionProperty` bindings, so I'll remove the WIP marker. :)

**Comment by chrisbobbe** [comment]
I've just rebased this past a171a051f, which did a lot of the work toward fixing #3540. I also made more commits out of the last few, so it should be more readable now. 🙂

File: src/api/subscriptions/setSubscriptionProperty.js
Comment (gnprice) [inline src/api/subscriptions/setSubscriptionProperty.js]: I do still like this consolidation! Even though it needs a bit of a wrinkle to maintain that "mute" vs. `in_home_view` facade mentioned in my previous comment.

I think a quick conditional within this function can take care of that reasonably cleanly, though. So this will take `is_muted` | `pin_to_top` | `push_notifications` as an argument, but if it's given `is_muted` it'll convert to `in_home_view` for what to actually pass to the server.

File: src/api/subscriptions/setSubscriptionProperty.js
Comment (gnprice) [inline src/api/subscriptions/setSubscriptionProperty.js:7]: Let's give this a quick bit of jsdoc that points to the API documentation:
https://chat.zulip.org/api/update-subscription-settings

That has quite a high ratio of helpfulness to effort, so it's a good minimal form of documentation for these API bindings. (And is really all they need when they don't have significant wrinkles of their own to document -- no sense duplicating what's in the server API docs.)

**Review by gnprice — COMMENTED** [review]
Thanks for the revision! Comments below.

File: src/message/messageActionSheet.js
Comment (chrisbobbe) [reply src/message/messageActionSheet.js:106]: Ooh, thanks for the catch—I think this was a rebase conflict-resolution error.

**Comment by chrisbobbe** [comment]
Thanks for the review! Revision pushed.

**Comment by chrisbobbe** [comment]
New revision pushed! 🙂

**Review by WesleyAC — APPROVED** [review]
Looks great, thanks for doing this!

**Comment by chrisbobbe** [comment]
Thanks for the reviews! Merged.

**Pull request merged**
